indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/SinglePhaseParallelIndexTaskRunner.java

```diff
@@ -343,6 +344,11 @@ public SegmentIdWithShardSpec allocateNewSegment(
       interval = granularitySpec.getSegmentGranularity().bucket(timestamp);
       version = ParallelIndexSupervisorTask.findVersion(versions, interval);
       if (version == null) {
+        final int maxAllowedLockCount = getIngestionSchema().getTuningConfig().getMaxAllowedLockCount();
+        if (maxAllowedLockCount >= 0 && locks.size() == maxAllowedLockCount) {
```

Comment:
Would change to `locks.size() == maxAllowedLockCount` just for safety

Reply:
Changed as suggested